File: frontend/src/types/comment.ts
// 评论基本信息接口
export interface Comment {
  id: number;
  content: string;
  post_id: number;
  author_id: number;
  author_name: string;
  created_at: string;
  updated_at?: string;
  parent_id?: number;
  like_count: number;
}

// 评论列表响应接口
export interface CommentList {
  comments: Comment[];
  total: number;
  page: number;
  page_size: number;
}

// 评论创建请求接口
export interface CommentCreateRequest {
  content: string;
  post_id: number;
  parent_id?: number;
} 

File: backend/app/api/responses/comment.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel
from datetime import datetime

from .user import UserInfoResponse

class CommentResponse(BaseModel):
    """评论响应模型
    
    用于API返回评论对象的标准格式
    """
    id: int
    content: str
    author_id: int
    post_id: int
    created_at: datetime
    updated_at: Optional[datetime] = None
    is_deleted: Optional[bool] = False
    
    # 可选关联字段，根据实际需要返回
    author: Optional[UserInfoResponse] = None
    
    class Config:
        from_attributes = True

class CommentListResponse(BaseModel):
    """评论列表响应模型"""
    comments: List[CommentResponse]
    total: int
    post_id: Optional[int] = None
    
    class Config:
        from_attributes = True

class CommentDeleteResponse(BaseModel):
    """评论删除响应模型"""
    id: int
    message: str = "评论已成功删除" 